### ISSUE TITLE
Missing info on Action.Http

### ISSUE THREAD

**Issue by evrentekin** [microsoft/AdaptiveCards#8009]
There's no documentation available on Action.Http

---
* URL: https://adaptivecards.io/explorer/Action.Submit.html

**Comment by fowl2** [comment]
There is some information here: https://learn.microsoft.com/en-us/outlook/actionable-messages/adaptive-card#actionhttp

**Comment by jwoo-msft** [comment]
Thanks @fowl2 
@evrentekin, we do not support Action.Http anymore. Please use Action.Submit and Action.Execute.
Outlook, however, may supports it; but they have moved away from it, and supports Action.Execute.

**Comment by mitquinn** [comment]
@jwoo-msft do you have an example of Action.Execute sending an http request?

**Comment by jwoo-msft** [comment]
@mitquinn 
https://learn.microsoft.com/en-us/adaptive-cards/authoring-cards/universal-action-model

**Comment by mitquinn** [comment]
> @mitquinn https://learn.microsoft.com/en-us/adaptive-cards/authoring-cards/universal-action-model

I feel like I am missing something. Where do you set the url, method, or headers?

**Comment by jwoo-msft** [comment]
@mitquinn,
Could you share your use case? The missing piece of information likely originates from the HostApp that hosts AdaptiveCards.

**Comment by mitquinn** [comment]
> @mitquinn, Could you share your use case? The missing piece of information likely originates from the HostApp that hosts AdaptiveCards.

Yeah absolutely.

So I was looking at this : https://powerautomate.microsoft.com/en-us/blog/how-to-use-actionable-messages-in-outlook-with-power-automate/

The general idea is that Power Automate is sending an adaptive card, which than is making an http request back to PA to handle the response.

In Step#2 they have this section: 

```json
{
  "type": "AdaptiveCard",
  "body": [
    {
      "type": "Container",
      "style": "emphasis",
      "items": [
        {
          "type": "TextBlock",
          "text": "Quick Poll",
          "wrap": true
        }
      ],
      "padding": "Default"
    },
    {
      "type": "Container",
      "id": "d96d06ec-2a33-c8cc-4698-430ae87f772f",
      "padding": "Default",
      "items": [
        {
          "type": "TextBlock",
          "size": "Medium",
          "weight": "Bolder",
          "text": "Take 2 minutes to help us plan the upcoming conference",
          "wrap": true
        }
      ],
      "spacing": "None",
      "separator": true
    },
    {
      "type": "Container",
      "id": "885220a9-5ab1-95dd-5b66-20f42c452fa9",
      "padding": "Default",
      "items": [
        {
          "type": "TextBlock",
          "weight": "Bolder",
          "text": "Where should we host the conference?",
          "wrap": true
        }
      ],
      "separator": true,
      "spacing": "None"
    },
    {
      "type": "Container",
      "id": "10017c5a-5ee9-46c5-537a-bdd9ab61225c",
      "padding": {
        "top": "None",
        "bottom": "Default",
        "left": "Default",
        "right": "Default"
      },
      "items": [
        {
          "type": "Input.ChoiceSet",
          "id": "Options",
          "spacing": "None",
          "placeholder": "Where should we host the conference?",
          "label": "Where should we host the conference?",
          "choices": [
            {
              "title": "Radisson",
              "value": "Radisson"
            },
            {
              "title": "Hilton",
              "value": "Hilton"
            },
            {
              "title": "Sheraton Downtown",
              "value": "Sheraton Downtown"
            },
            {
              "title": "W Downtown",
              "value": "W Downtown"
            },
            {
              "title": "Clarks",
              "value": "Clarks"
            }
          ],
          "style": "expanded"
        }
      ],
      "spacing": "None"
    },
    {
      "type": "Container",
      "spacing": "None",
      "items": [
        {
          "type": "TextBlock",
          "text": "Other:",
          "wrap": true,
          "horizontalAlignment": "Left"
        }
      ],
      "padding": {
        "top": "None",
        "bottom": "None",
        "left": "Default",
        "right": "Default"
      }
    },
    {
      "type": "Container",
      "id": "3734939f-dd4d-79be-cf3c-2ad791f382ba",
      "padding": {
        "top": "Default",
        "bottom": "None",
        "left": "Default",
        "right": "Default"
      },
      "items": [
        {
          "type": "Input.Text",
          "id": "Suggestions",
          "placeholder": "Have another suggestion?",
          "label": "Other:"
        }
      ],
      "spacing": "None"
    },
    {
      "type": "Container",
      "id": "a27409f1-a48e-982f-8e26-594bdd1468e8",
      "padding": "Default",
      "items": [
        {
          "type": "ActionSet",
          "horizontalAlignment": "Left",
          "actions": [
            {
              "type": "Action.Http",
              "title": "Submit",
              "method": "POST",
              "url": "https://prod-00.westus.logic.azure.com:443/workflows/.... this is the URL that you took from the flow",
              "body": "{\"Option\": {{\"Options.value\"}}, \"SuggestionText\": {{\"Suggestions.value}}\"}",
              "headers": [
                {
                  "name": "Authorization",
                  "value": ""
                },
                {
                  "name": "Content-Type",
                  "value": "application/json"
                }
              ],
              "isPrimary": true,
              "style": "positive"
            }
          ]
        }
      ],
      "spacing": "None"
    }
  ],
  "$schema": "https://adaptivecards.io/schemas/adaptive-card.json",
  "version": "1.0",
  "padding": "None",
  "originator": "75df... this ID can be taken from the form that you submitted"
}
```

But the important part there is specifically: 

```json
"actions": [
            {
              "type": "Action.Http",
              "title": "Submit",
              "method": "POST",
              "url": "https://prod-00.westus.logic.azure.com:443/workflows/.... this is the URL that you took from the flow",
              "body": "{\"Option\": {{\"Options.value\"}}, \"SuggestionText\": {{\"Suggestions.value}}\"}",
              "headers": [
                {
                  "name": "Authorization",
                  "value": ""
                },
                {
                  "name": "Content-Type",
                  "value": "application/json"
                }
              ],
              "isPrimary": true,
              "style": "positive"
            }
          ]
```

Everything works great in version 1.0. But if you tried to update the adaptive card beyond v1.0 it all stops working.

The weird thing is that the card actually renders correctly in v1.4 except the Action.Http button doesn't appear. I am assuming it needs to be updated to Action.Execute but there are no params for setting url, method, headers, etc. From reading the docs it seems like the Action.Http method is no longer supported and is replaced by Action.Execute but Action.Execute offers no way to make an http request.

**Comment by mitquinn** [comment]
@cc13com unfortunately no. I tried a bunch of different configurations with no luck. 

I saw this in the FAQ which seems to point at the right direction: [Upgrading from 1.0 to 1.4](https://learn.microsoft.com/en-us/outlook/actionable-messages/actionable-messages-faq#i-have-upgraded-the-adaptive-card-version-from-10-to-14-and-my-action-buttons-have-disappeared)

Digging a bit deeper I found this: [Backwards Compatability](https://learn.microsoft.com/en-us/outlook/actionable-messages/universal-action-model#backward-compatibility)

It just seems like its not possible to use Action.Execute to trigger any kind of http request. Which is super disappointing. They mention using Bots to handling the functionality but that doesn't really seem like an alternative to me.
